### PR TITLE
String Reading and Path Typing

### PR DESCRIPTION
**Pull request by H00N24**
Hi in this PR I would like to add some functionality to the Python wrapper:

- `parse_string` function that can parse already loaded string as in the cloud setup we're loading this file from zip in S3 thus we already have it in memory
- adds `Union[str | Pathlike]` typing to `parse` to allow str paths for  mypy type-checking + mypy CI check
- (optional) nix & direnv setup for local dev. For users using these tools it automatically builds an environment will all dev dependencies.

I've built it on top of the other PR just so I could pass the Rust cargo test exception, we can rebase it on top of the master if we decide to change the previous PR.

### CODE REVIEW

File: src/python_wrapper/src/lib.rs [hunk 1]
```diff
@@ -27,19 +27,15 @@ fn handle_io_error(path: &Path, e: std::io::Error) -> PyErr {
...
 /// - `RuntimeError` if source code parsing failed.
 #[pyfunction]
-fn parse(py: Python, path: PathBuf) -> PyResult<Vec<CvlElementPy>> {
-    let src = file_contents(path.as_path())?;
-
+fn parse_string(py: Python, src: String) -> PyResult<Vec<CvlElementPy>> {
     let elements = Builder::new(&src).build().map_err(|_| {
-        let display = path.display();
-        let desc = format!("failed to parse source file: {display}");
+        let desc = String::from("failed to parse source code");
```
Comment (pickx) [inline src/python_wrapper/src/lib.rs:38]: nit, can be reduced to
```rust
.map_err(|_| PyRuntimeError::new_err("failed to parse source code"))?;
```

File: .envrc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+if command -v nix &>/dev/null; then
+    # Load .envrc from certora dir if exists
+    [ -f ../.envrc ] && source_env ..
+    # Use nix if it's available
+    use flake
+fi
```
Comment (pickx) [inline .envrc:6]: so to be clear, what's the behavior if `nix` is unavailable?

File: tests/simple.py [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+
+assert len(p_file) == 3, "should parse to 3 elements"
+
+p_file_as_string = cvldoc_parser.parse(spec_file.as_posix())
+p_from_string = cvldoc_parser.parse_string(spec_file.read_text())
+
+p = map(as_list, [p_file, p_file_as_string, p_from_string])
+assert all(
+    x == y for x in p for y in p
+), "all three parsers should parse the same elements"
```
Comment (pickx) [inline tests/simple.py:32]: 💯

**Review by pickx — COMMENTED** [review]
can't say I have any experience with nix but not much to complain about here.

File: src/python_wrapper/src/lib.rs (hunk 1 above)
Comment (H00N24) [reply src/python_wrapper/src/lib.rs]: Fixed.

File: .envrc (hunk 2 above)
Comment (H00N24) [reply .envrc]: So if the nix is not available `direnv` will only create a virtual environment using the stdlib `venv` package and activate the virtual environment.

**Review by pickx — APPROVED** [review]

**Pull request merged**